brigade.js

```diff
@@ -42,6 +83,45 @@ function buildAndPublishImages(project, version) {
   return job;
 }
 
+// handleIssueComment handles an issue_comment event, parsing the comment text
+// and determining whether or not to trigger an action
+function handleIssueComment(e, p) {
+  payload = JSON.parse(e.payload);
+
+  // Extract the comment body and trim whitespace
+  comment = payload.body.comment.body.trim();
+
+  // Here we determine if a comment should provoke an action
+  switch(comment) {
+    case "/brig run":
+      runSuite(e, p);
+      break;
+    default:
+      console.log(`No applicable action found for comment: ${comment}`);
+  }
+}
+
+// checkRequested is the default function invoked on a check_run:* event
+//
+// It determines which check is being requested (from the payload body)
+// and runs this particular check, or else throws an error if the check
+// is not found
+function checkRequested(e, p) {
+  payload = JSON.parse(e.payload);
+
+  // Extract the check name
+  name = payload.body.check_run.name;
+
+  // Determine which check to run
+  switch(name) {
+    case "tests":
+      runTests(e, p).catch(err => {console.error(err.toString())});
```

Comment:
Don't catch the error (let it propagate), but _do_ return the promise that is returned from the call to `runTests()`.

Reply:
👍 updated.